crates/pallet-feeds/src/feed_processor.rs

```diff
@@ -15,17 +15,82 @@
 
 //! Defines FeedProcessor and its types
 
+use crate::CallObject;
+use codec::{Compact, Decode};
 use sp_runtime::{DispatchError, DispatchResult};
 use sp_std::{vec, vec::Vec};
-use subspace_core_primitives::{crypto, Sha256Hash};
+use subspace_core_primitives::Sha256Hash;
 
-/// Object mapping that points to an object in a block
+/// Holds the offset to some portion of data within/or the object
 #[derive(Debug)]
-pub struct FeedObjectMapping {
-    /// Key scoped to the feed
-    pub key: Sha256Hash,
-    /// Offset of the data within object
-    pub offset: u32,
+pub enum FeedObjectMapping {
+    /// Maps the entire object treating it as content
+    /// If key is provided, it is namespaced to feed to avoid collisions
+    Object { key: Option<Vec<u8>> },
+    /// Maps some data within the object at the specific offset.
+    /// Data must be a scale encoded `Vec<u8>`
+    /// If key is provided, it namespaced to feed to avoid collisions
+    Content { key: Option<Vec<u8>>, offset: u32 },
+}
+
+impl FeedObjectMapping {
+    pub(crate) fn try_into_call_object<KNS, CO, CH>(
+        self,
+        object: &[u8],
+        key_namespace: KNS,
+        call_offset: CO,
+        content_hasher: CH,
+    ) -> Option<CallObject>
+    where
+        KNS: Fn(&[u8]) -> Sha256Hash,
+        CO: Fn(u32, bool) -> u32,
+        CH: Fn(&[u8]) -> Sha256Hash,
```

Comment:
I feel like this API it too cumbersome. It should just take an object reference and another argument (enum) that defines hashing function, whose implementation should be supported by runtime (with SHA-256 to start and adding Blake2 and others in the future).

With suggested enum above there will be just 2 branches:
* one reads a vector and hashes content
* another combines feed ID with custom key and hashes them
* in both cases offset needs to be checked for sanity